spec/src/test/java/impl/UserResourceTest.java

```diff
@@ -94,23 +124,26 @@ public void shouldFetchExistingUserInDatabaseIfReturningUser() {
 
     @Test
     public void shouldAddCookieToHeader() {
-        // given
+        // given a openId user
         User user = insertUser();
         ImmutableOpenIdToken openIdObject = new ImmutableOpenIdToken(user.getName(), user.getEmail(), "pictureUrl");
         when(openIdValidator.validate(any())).thenReturn(just(openIdObject));
 
-        // when
-        User returnedUser = userResource.generateToken(OPEN_ID_TOKEN).toBlocking().singleOrDefault(null);
+        // when the user signs in.
+        User returnedUser = userResource.signIn(OPEN_ID_TOKEN).toBlocking().singleOrDefault(null);
 
-        // then
+        // then a secure cookie should be created
         ArgumentCaptor<Map> mapArgumentCaptor = ArgumentCaptor.forClass(Map.class);
         assertNotNull(returnedUser);
         verify(responseHeaderHolder, times(1)).addHeaders(any(), mapArgumentCaptor.capture());
         String setCookieHeader = (String) mapArgumentCaptor.getValue().get("Set-Cookie");
-        assertTrue(setCookieHeader.contains("application="));
-        assertTrue(setCookieHeader.contains("application="));
-        assertTrue(setCookieHeader.contains("; path=/; domain=.rocket-fuel;"));
-
+        assertThat(setCookieHeader).contains("application=ey");
+        assertThat(setCookieHeader).contains("Path=/;");
+        assertThat(setCookieHeader).contains("Domain=.rocket-fuel;");
```

Comment:
".rocket-fuel" is hard coded in serveral places, isn't applicationTokenConfig a better data source to check against?
I believe 3600 comes from a constant in UserResourceImpl as well.